rust/rope/src/subset.rs

```diff
@@ -48,57 +48,54 @@ impl SubsetBuilder {
 
     pub fn build(mut self) -> Subset {
         if self.e > self.b {
-            self.dels.push((self.b, self.e));
+            self.ranges.push((self.b, self.e));
         }
-        Subset(self.dels)
+        Subset(self.ranges)
     }
 }
 
 impl Subset {
     // mostly for testing
-    pub fn apply_to_string(&self, s: &str) -> String {
+    pub fn delete_in_string(&self, s: &str) -> String {
         let mut result = String::new();
-        for (b, e) in self.range_iter(s.len()) {
+        for (b, e) in self.complement_iter(s.len()) {
             result.push_str(&s[b..e]);
         }
         result
     }
 
     // Maybe Subset should be a pure data structure and this method should
     // be a method of Node.
-    pub fn apply<N: NodeInfo>(&self, s: &Node<N>) -> Node<N> {
+    pub fn delete_in<N: NodeInfo>(&self, s: &Node<N>) -> Node<N> {
```

Comment:
This would benefit from documentation.

I also prefer `delete_from` to `delete_in`.